e2e_test/e2e_test.go

```diff
@@ -36,3 +38,54 @@ func TestSendCelo(t *testing.T) {
 	err = network.AwaitTransactions(ctx, tx)
 	require.NoError(t, err)
 }
+
+func TestSingleNodeManyTxs(t *testing.T) {
+	rounds := 100
+	txsPerRound := 5
+	accounts := test.Accounts(1)
+	gc := test.GenesisConfig(accounts)
+	gc.Istanbul.Epoch = uint64(rounds) * 50 // avoid the epoch for this test
+	network, err := test.NewNetwork(accounts, gc)
+	require.NoError(t, err)
+	defer network.Shutdown()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	for r := 0; r < rounds; r++ {
+		txs := make([]*types.Transaction, 0, txsPerRound)
+		for j := 0; j < txsPerRound; j++ {
+			tx, err := network[0].SendCelo(ctx, common.Address{}, 1)
+			require.NoError(t, err)
+			require.NotNil(t, tx)
+		}
+		err = network.AwaitTransactions(ctx, txs...)
+		require.NoError(t, err)
+	}
+}
+
+func TestManyNodes(t *testing.T) {
+	accounts := test.Accounts(4)
+	gc := test.GenesisConfig(accounts)
+	rounds := 5
+	txsPerNodePerRound := 3
+	gc.Istanbul.Epoch = 10000                               // avoid the epoch
+	network, err := test.NewConcurrentNetwork(accounts, gc) // start concurrently all nodes
+	require.NoError(t, err)
+	defer network.Shutdown()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer cancel()
+
+	for r := 0; r < rounds; r++ {
+		txs := make([]*types.Transaction, 0, len(network)*txsPerNodePerRound)
+		for i := range network {
+			for j := 0; j < txsPerNodePerRound; j++ {
+				tx, err := network[i].SendCelo(ctx, network[(i+1)%len(network)].Address, 1)
+				require.NoError(t, err)
+				require.NotNil(t, tx)
+				txs = append(txs, tx)
+			}
+		}
+		err = network.AwaitTransactions(ctx, txs...)
+		require.NoError(t, err)
+	}
+}
```

Comment:
This test fails fairly regularly for me it was just exiting with a critical log, so I added a panic into `os.Exit` to help pinpoint where this was happening and the result was this.

```
CRIT [09-13|14:59:21.583] Failed to store updated uptime           err="leveldb: closed"
----------------------------------------- someone called exit goroutine 3345 [running]:
os.Stack(0x1de02e0, 0xa2f850, 0x1ba0cb8)
	/usr/local/go/src/os/proc.go:62 +0x9f
os.Exit(0x1)
	/usr/local/go/src/os/proc.go:76 +0x26
github.com/celo-org/celo-blockchain/log.Crit(0x13f90e8, 0x1e, 0xc00cde54f8, 0x2, 0x2)
	/home/pierspowlesland/projects/celo-blockchain/log/root.go:60 +0x7e
github.com/celo-org/celo-blockchain/core/rawdb.WriteAccumulatedEpochUptime(0x7ffab807b330, 0xc017250af8, 0x1, 0xc020cd14c0)
	/home/pierspowlesland/projects/celo-blockchain/core/rawdb/accessors_chain.go:537 +0x1b1
github.com/celo-org/celo-blockchain/consensus/istanbul/uptime/store.(*uptimeStoreImpl).WriteAccumulatedEpochUptime(0xc012684080, 0x1, 0xc020cd14c0)
	/home/pierspowlesland/projects/celo-blockchain/consensus/istanbul/uptime/store/store.go:23 +0x71
github.com/celo-org/celo-blockchain/consensus/istanbul/uptime.(*Monitor).ProcessBlock(0xc00cde5ac0, 0xc00d4ece80, 0x2, 0x2)
	/home/pierspowlesland/projects/celo-blockchain/consensus/istanbul/uptime/monitor.go:133 +0x3a3
github.com/celo-org/celo-blockchain/core.(*BlockChain).insertPreprocessedBlock(0xc013515200, 0xc00d4ece80, 0xc01e734740, 0x4, 0x4, 0x0, 0x0, 0x0, 0xc0023fc580, 0xc000b0d401, ...)
	/home/pierspowlesland/projects/celo-blockchain/core/blockchain.go:1492 +0x42f
github.com/celo-org/celo-blockchain/core.(*BlockChain).InsertPreprocessedBlock(0xc013515200, 0xc00d4ece80, 0xc01e734740, 0x4, 0x4, 0x0, 0x0, 0x0, 0xc0023fc580, 0x0, ...)
	/home/pierspowlesland/projects/celo-blockchain/core/blockchain.go:1472 +0x1e5
github.com/celo-org/celo-blockchain/miner.(*worker).start.func2(0xc00d4ece80, 0xc01e734740, 0x4, 0x4, 0x0, 0x0, 0x0, 0xc0023fc580)
	/home/pierspowlesland/projects/celo-blockchain/miner/worker.go:187 +0xb3
created by github.com/celo-org/celo-blockchain/consensus/istanbul/backend.(*Backend).Commit
	/home/pierspowlesland/projects/celo-blockchain/consensus/istanbul/backend/backend.go:539 +0x634
```

We see that the error is "leveldb: closed", my guess is that we have called `Close` on the node but that there is no synchronization between that and the goroutine writing uptime. I'm happy to dig in to this further if you would like.